### ISSUE TITLE
Remove deprecated SSL settings from Logstash core

### ISSUE THREAD

**Issue by robbavey** [elastic/logstash#16652]
- ~[ ] Enumerate deprecated SSL settings in core~
- ~[ ] Ensure that all unit and integration tests still pass~
- ~[ ] Add details of change to the 9.0 breaking changes documentation in core.~

See https://github.com/elastic/logstash/issues/16652#issuecomment-2549812346 for write up, but I was unable to identify any actions required for the steps listed in this ticket.

**Comment by donoghuc** [comment]
In order to get the shape and scope of this I compiled a list of all the deprecates ssl options i've seen so far in the plugin work and searched a copy of a logstash checkout. There are obviously quite a few false positives (ssl is going to be a very popular one 😅 ). I've attached the hist and where they appear. I will be going through them and first removing obvious false positives. I'll then move on to classifying/organising the rest of the occurances. Let me know if i'm way over-thinking this though! [matches.txt](https://github.com/user-attachments/files/18133043/matches.txt)

**Comment by robbavey** [comment]
That definitely seems like a fair few false positives... We can skip anything modules related.

What we want to do here is ensure that we:
* Remove any references to obsolete SSL settings in tests, docs, or code where we map settings to plugins 
  * Any Logstash configuration settings from `logstash.yml` or docker config, for functionality like the configuration of internal collection based monitoring and central pipeline management are correctly mapped to non-deprecated settings.
* Remove any settings that have previously been previously deprecated in Logstash core that could now be marked as obsolete.
  *  This may be a no-op - [the work](https://github.com/elastic/logstash/pull/15045) to replace deprecated settings did not, AFAICT, mark any SSL-related settings in Logstash core as deprecated  - but I would like to get verification.
  * We should not mark any SSL settings as `:obsolete` that were not previously marked as deprecated. 
* Any changes that we do make that are external facing should be put in the breaking changes doc

**Comment by donoghuc** [comment]
> * Remove any references to obsolete SSL settings in tests, docs, or code where we map settings to plugins 
>    * Any Logstash configuration settings from `logstash.yml` or docker config, for functionality like the configuration of internal collection based monitoring and central pipeline management are correctly mapped to non-deprecated settings.

The part I originally missed is that the *keys* in the mappings (IE the value we set when configuring these on the LS side) are not changing, its the *translation* of those into the settings used by the underlying plugins. I went through and confirmed that everywhere we need to do a translation is now doing so with modern/standardized ssl settings for plugins. 
> * Remove any settings that have previously been previously deprecated in Logstash core that could now be marked as obsolete.

I was unable to find any plugin ssl settings that are carrying a deprecation warnings in LS. 
  *  This may be a no-op - [the work](https://github.com/elastic/logstash/pull/15045) to replace deprecated settings did not, AFAICT, mark any SSL-related settings in Logstash core as deprecated  - but I would like to get verification.

Verified the linkede PR was complete and no deprecations need to be promoted to obsolete/removed. (Thank you BTW for linking this PR as this was the key to me understanding the scope of this work). 

>  * We should not mark any SSL settings as `:obsolete` that were not previously marked as deprecated. 

This is a noop
* Any changes that we do make that are external facing should be put in the breaking changes doc

I did not find any external facing deprecations/removals to document. 

My overall findings for this ticket is that it is resulting in a noop. It is always hardest to prove a negative, but i'm just not finding anything that needs to be removed as part of 9.0.0 relating to ssl setting standardization.
